ui/packages/shared/profile/src/ProfileIcicleGraph/IcicleGraphArrow/IcicleGraphNodes.tsx

```diff
@@ -197,6 +213,23 @@ export const IcicleNode = React.memo(function IcicleNodeNoMemo({
   const cumulative = cumulativeColumn?.get(row) !== null ? BigInt(cumulativeColumn?.get(row)) : 0n;
   const diff: bigint | null = diffColumn?.get(row) !== null ? BigInt(diffColumn?.get(row)) : null;
   const childRows: number[] = Array.from(table.getChild(FIELD_CHILDREN)?.get(row) ?? []);
+  const [highlightSimilarStacksPreference] = useUserPreference<boolean>(
+    USER_PREFERENCES.HIGHLIGHT_SIMILAR_STACKS.key
+  );
+  const [currentColorPalette] = useUserPreference<string>(
+    USER_PREFERENCES.FLAMEGRAPH_COLOR_PROFILE.key
+  );
+
+  const shouldBeHighlightedIfSimilarStacks = useMemo(() => {
+    return (
+      highlightSimilarStacksPreference &&
+      functionName != null &&
+      functionName === hoveringName &&
+      row !== hoveringRow
+    );
+  }, [highlightSimilarStacksPreference, functionName, row, hoveringName, hoveringRow]);
```

Comment:
I guess, the performance impact is because of this: hovering around nodes is going to trigger this memo and cause a re-render on all the nodes in the flamegraph. 

Maybe pre-computing the nodes to highlight might help?

Reply:
Debouncing in a way that this computation runs only when the mouse rests on a node for a second (or half) could help smoothen it, too.

Reply:
thanks for the suggestions, i'll give them a go.